crates/bootstrap_client/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+//! A client for the Kitsune2 bootstrap server.
+
+#![deny(missing_docs)]
+
+use kitsune2_api::{AgentInfoSigned, DynVerifier, K2Error, K2Result, SpaceId};
+use std::sync::Arc;
+
+/// Send the agent info, for the given space, to the bootstrap server.
+pub fn put(
+    server_url: Arc<str>,
+    agent_info: Arc<AgentInfoSigned>,
+) -> K2Result<()> {
+    let url = format!(
+        "{server_url}/bootstrap/{}/{}",
+        &agent_info.space, &agent_info.agent
+    );
+
+    let encoded = agent_info.encode()?;
+    ureq::put(&url)
+        .send_string(&encoded)
+        .map_err(|e| K2Error::other_src("Failed to put agent info", e))?;
+
+    Ok(())
+}
+
+/// Get all agent infos from the bootstrap server for the given space.
+pub fn get(
+    server_url: Arc<str>,
+    space: SpaceId,
+    verifier: DynVerifier,
```

Comment:
Require a verifier so that the client can only be used to retrieve agent infos that are considered valid by the provided verifier. Of course the user can provide a bad verifier but at least they're encouraged to think about what they're getting back.